py-polars/polars/_cpu_check.py

```diff
@@ -1,13 +1,249 @@
+# Vendored parts of the code from https://github.com/flababah/cpuid.py,
+# so we replicate its copyright license.
+
+# Copyright (c) 2014 Anders Høst
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+from __future__ import annotations
+
+import ctypes
+import os
+from ctypes import CFUNCTYPE, POINTER, c_long, c_size_t, c_uint32, c_ulong, c_void_p
+from typing import ClassVar
+
 """
 Determine whether Polars can be run on the current CPU.
 
-This must be done in pure Python, before the Polars binary is imported.
+This must be done in pure Python, before the Polars binary is imported. If we
+were to try it on the Rust side the compiler could emit illegal instructions
+before/during the CPU feature check code.
 """
-# Set to True during build process if Polars was built with a limited instruction set
-_LTS_CPU = False
 
+# Replaced during the build process with the architecture of our wheel.
+_POLARS_ARCH = "unknown"
```

Comment:
I think this technically can be a boolean since this whole check only makes sense for `x86-64`. But I guess it's fine to keep it this way.